src/main/java/games/strategy/triplea/util/UnitCategory.java

```diff
@@ -83,24 +84,20 @@ private void createDependents(final Collection<Unit> dependents) {
 
   @Override
   public boolean equals(final Object o) {
-    if (o == null) {
-      return false;
-    }
-    if (!(o instanceof UnitCategory)) {
+    if (o == null || !(o instanceof UnitCategory)) {
```

Comment:
The `o == null` expression can just be removed.  `o instanceof UnitCategory` will evaluate to `false` if `o` is `null`.